pymc/model/core.py

```diff
@@ -1352,7 +1352,7 @@ def make_obs_var(
 
             if test_value is not None:
                 # We try to reuse the old test value
-                rv_var.tag.test_value = np.broadcast_to(test_value, rv_var.shape)
+                rv_var.tag.test_value = np.broadcast_to(test_value, rv_var.type.shape)
             else:
                 rv_var.tag.test_value = data
```

Comment:
There could be None in `type.shape`. Broadcasting shouldn't be needed because the RV should always have the right shape.

I suggest simplifying the logic to:
```
if test_value is None:
    rv_var.tag.test_value = data
```
And ignore the whole broadcast logic

Reply:
I can't think of a scenario where this could be None though because the RV is created by the time this method is called. Can we just remove this section al together?

Reply:
I implemented the first suggestion.

I can't see a case where `test_value` is None either, but you're much more familiar with this code than I am so I'll implement the option you prefer! :) Should I just remove the `if pytensor.config.compute_test_value != "off":` block completely?

Reply:
Yes, let's do that and see if any tests fails

Reply:
Done!